### PR TITLE
Adds toggle port security on network create

### PR DESCRIPTION
**Pull request by snowjet**
##### SUMMARY
Added a new property, 'port_security_enabled' which is a boolean to
enable or disable port security during network creation. The default
behaviour will enable port security, security group and anti spoofing
will act as before. When the attribute is set to False, security
group and anti spoofing are disabled on the ports created on this
network.

Utilises this commit in the shade library - https://github.com/openstack-infra/shade/commit/5b81959b32c52b37ff1465cc3a8d39c8b3a47dee

Its in master so should be available in shade versions > 1.27.1

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
cloud / openstack / os_network 

##### ADDITIONAL INFORMATION
Example use of the os_network module using the new property port_security_enabled:
```
- os_network:
    cloud: mycloud
    state: present
    name: priv_network
    port_security_enabled: false  
```

### CODE REVIEW

**Comment by snowjet** [comment]
ready_for_review

**Comment by horacimacias** [comment]
any chance of having this done for os_port as well?
see https://github.com/ansible/ansible/issues/24694

**Comment by snowjet** [comment]
@horacimacias - It will need to be added to the shade module first.  https://github.com/openstack-infra/shade/blob/master/shade/openstackcloud.py

Once I have this merged, I am happy to see if I can add port create/update into the shade module.

**Comment by snowjet** [comment]
Hi  @emonty  @Shrews  - this is the other half of the change from - https://review.openstack.org/#/c/550646/ . Is there any feedback regarding this pull request? Just looking for support getting it merged.

**Comment by rjet** [comment]
ready_for_review

**Comment by snowjet** [comment]
ready_for_review

**Comment by snowjet** [comment]
bot_status

File: lib/ansible/modules/cloud/openstack/os_network.py
Comment (tbreeds) [inline lib/ansible/modules/cloud/openstack/os_network.py]: You need to raise this to 1.27.1 to match your commit message.

Comment (snowjet) [reply lib/ansible/modules/cloud/openstack/os_network.py]: Thanks. I missed that. Committed and pushed.

**Comment by snowjet** [comment]
ready_for_review

File: lib/ansible/modules/cloud/openstack/os_network.py
Comment (mnaser) [inline lib/ansible/modules/cloud/openstack/os_network.py]: Can you change this to 2.8?

File: lib/ansible/modules/cloud/openstack/os_network.py
Comment (mnaser) [inline lib/ansible/modules/cloud/openstack/os_network.py]: Please remove the shade dependency, it is no longer required.

File: lib/ansible/modules/cloud/openstack/os_network.py
Comment (mnaser) [inline lib/ansible/modules/cloud/openstack/os_network.py]: https://github.com/openstack/openstacksdk/blob/master/openstack/cloud/openstackcloud.py#L3382-L3386

`availability_zone_hints` comes after `project_id` so you need to fix the order, or use a keyword argument.

File: lib/ansible/modules/cloud/openstack/os_network.py
Comment (mnaser) [inline lib/ansible/modules/cloud/openstack/os_network.py]: Same here, I suggest just adding `port_security_enabled=port_security_enabled` or something like that.  In this case, `project_id` is pointing to `port_security_enabled` value.

**Review by mnaser — CHANGES_REQUESTED** [review]

**Comment by mnaser** [comment]
needs_revision

File: lib/ansible/modules/cloud/openstack/os_network.py
Comment (tbreeds) [reply lib/ansible/modules/cloud/openstack/os_network.py]: done.

File: lib/ansible/modules/cloud/openstack/os_network.py
Comment (tbreeds) [reply lib/ansible/modules/cloud/openstack/os_network.py]: done.

File: lib/ansible/modules/cloud/openstack/os_network.py
Comment (tbreeds) [reply lib/ansible/modules/cloud/openstack/os_network.py]: done.

File: lib/ansible/modules/cloud/openstack/os_network.py
Comment (tbreeds) [reply lib/ansible/modules/cloud/openstack/os_network.py]: done.

**Comment by tbreeds** [comment]
ready_for_review

**Comment by mnaser** [comment]
thanks @tbreeds

**Comment by mnaser** [comment]
shipit

**Comment by tbreeds** [comment]
Thanks!   What happens now?

**Comment by mnaser** [comment]
@tbreeds need one more "maintainer" to hit the "shipit" button and you'll be in :)

File: lib/ansible/modules/cloud/openstack/os_network.py
Comment (odyssey4me) [inline lib/ansible/modules/cloud/openstack/os_network.py]: Should this default to `False` instead of `None` to match the default in openstacksdk?

Comment (jamescassell) [reply lib/ansible/modules/cloud/openstack/os_network.py]: No, it should be `True`

Edit: this option should default to true to be consistent with current behavior and good security practice.

Comment (514amir) [reply lib/ansible/modules/cloud/openstack/os_network.py]: @tbreeds could you make that small fix so it can be merged please!

**Comment by cloudnull** [comment]
shipit

**Review by cloudnull — APPROVED** [review]

**Review by odyssey4me — APPROVED** [review]

**Comment by odyssey4me** [comment]
ship

**Comment by odyssey4me** [comment]
shipit

**Pull request merged**
